packages/gatsby-source-workspace/gatsby-node.js

```diff
@@ -0,0 +1,62 @@
+const getWorkspaces = require("get-workspaces").default;
+const findUp = require("find-up");
+
+const path = require("path");
+const { createContentDigest } = require("gatsby-core-utils");
+/*
+I'm mostly just implementing the workspaces as it exists. One change, I've renamed 'config'
+to 'meta' - this might still be the wrong name but I want to try it out.
+
+(meta represents the full contents of the package.json)
+*/
+
+const badFindRepoRoot = async () => {
+  let badPath = await findUp(".git/index");
+  return path.resolve(badPath, "..", "..");
+};
+
+exports.createSchemaCustomization = ({ actions }) => {
+  let { createTypes } = actions;
+
+  let typeDefs = `
+      type workspaceInfo implements Node {
+        name: String
+        dir: String
+        relativeDir: String
+        meta: JSON
+        version: String
+      }
+    `;
+  createTypes(typeDefs);
+};
+
+exports.sourceNodes = async (
+  { actions, createNodeId, options },
+  pluginOptions
+) => {
+  let { createNode } = actions;
+  let cwd = pluginOptions.cwd || (await badFindRepoRoot());
```

Comment:
lol, I wrote a thing to do this, it's in this repo, `find-workspaces-root`.

Reply:
:P I found it in my second round of editing on this (and implementing a slightly more opinionated extra layer)

Might move this to a draft PR